boards/doc.txt

```diff
@@ -17,6 +17,55 @@
  * device drivers should go into this module.
  *
 
+Board Selection Guide {#board-selection-guide}
+=====================
+
+There is no single best board to buy for use with RIOT, as hardware design
+involves many trade-offs. The challenge is to clearly asses the use case and
+collect the requirements and to find the best match for those.
+
+With this in mind, it is still possible to give some hints and recommendations:
+
+1. During Development:
+    - An integrated programmer/debugger is immensely useful. At the very least
+      a standard JTAG/SWD connector for quick and fool-proof connection is
+      required
+    - If possible, pick an MCU with *more* RAM and flash than your target board.
+      This allows to spend the excess in memory for debug features such as
+      utility modules, verbose assert messages, etc. during development
+    - Standard connectors such as Arduino Headers, or Grove, STEMMA, STEMMA QT,
+      QUIIC connectors etc. make hardware prototyping easier
+2. Final Product:
+    - An integrated programmer/debugger is a waste of resources (money, power,
+      space, ...)
+    - ...
+
+Popular Boards
+--------------
+
+| Board             | Use Case      | Performance   | Integrated Debugger   | Integrated Networking             | Arduino Headers   | Other Connectors  | Integrated Sensors/...                            | Breadboard Friendly   | Costs         | Remarks                       |
+|:----------------- |:------------- |:------------- |:--------------------- |:--------------------------------- |:----------------- |:----------------- |:------------------------------------------------- |:--------------------- |:------------- |:----------------------------- |
+| arduino-due       | Education     | ++            | ✔                     | ✗                                 | +++ (Uno, Mega)   | None              | - (1 LED)                                         | +                     | o             | Better buy the nrf52840dk     |
+| arduino-mega2560  | Legacy Shit   | --            | ✗                     | ✗                                 | +++ (Uno, Mega)   | None              | - (1 LED)                                         | +                     | --            | Better buy the nrf52840dk     |
+| arduino-uno       | Legacy Shit   | ---           | ✗                     | ✗                                 | ++  (Uno)         | None              | - (1 LED)                                         | +                     | --            | You like pain, don't you?     |
```

Comment:
> Legacy Shit

I certainly don't want a fun project to become all suits, but I'd still pick a different wording. "Legacy"? "Because you have them"?